### PR TITLE
OCPBUGS-45971: e2e: tuned degraded test fix

### PR DESCRIPTION
**Pull request by rbaturov**
* Addressed a corner case where nodes based on VMs would always log a warning without verifying if the profile was degraded. This incorrect behavior has been fixed to properly check the profile state before logging warnings.
* Modified isVM func to execute "systemd-detect-virt" command after chroot to rootfs. This will ensures accurate virtualization detection by running the command in the host's context instead of the container.

### CODE REVIEW

File: test/e2e/performanceprofile/functests/1_performance/performance.go [hunk 1]
```diff
@@ -128,12 +128,16 @@ var _ = Describe("[rfe_id:27368][performance]", Ordered, func() {
...
+				isNodeBasedOnVM, err := infrastructure.IsVM(context.TODO(), &node)
 				Expect(err).ToNot(HaveOccurred(), "Failed to detect if the node is based on VM")
-				if isNodeBasedOnVM {
-					testlog.Warning(fmt.Sprintf("Tuned profile is degraded. A warning raised as the node is based on a VM. Error message: %s", degradedCondition.Message))
-				} else {
-					Expect(degradedCondition.Status).To(Equal(corev1.ConditionFalse), "Tuned profile is degraded. Error message: %s", degradedCondition.Message)
+
+				if degradedCondition.Status == corev1.ConditionTrue {
+					message := fmt.Sprintf("Tuned profile is degraded. Error message: %s", degradedCondition.Message)
+					if isNodeBasedOnVM {
```
Comment (jmencak) [inline test/e2e/performanceprofile/functests/1_performance/performance.go:136]: Nit: I know the previous code had the same "issue", but perhaps using the "happy path" and inverting the condition might  make the code a bit more readable?

File: test/e2e/performanceprofile/functests/utils/infrastructure/vm.go [hunk 2]
```diff
@@ -10,11 +10,11 @@ import (
 )
 
 // IsVM checks if a given node's underlying infrastructure is a VM
-func IsVM(node *corev1.Node) (bool, error) {
+func IsVM(ctx context.Context, node *corev1.Node) (bool, error) {
```
[the remark this answers is not in the document]
Comment (rbaturov) [reply test/e2e/performanceprofile/functests/utils/infrastructure/vm.go:13]: I should have used it.

File: test/e2e/performanceprofile/functests/1_performance/performance.go (hunk 1 above)
Comment (rbaturov) [reply test/e2e/performanceprofile/functests/1_performance/performance.go:136]: If `isNotBasedOnVM` you mean?
Not sure I understood

Comment (jmencak) [reply test/e2e/performanceprofile/functests/1_performance/performance.go:136]: Actually, looking more closely, it looks fine as is. Sorry for the noise.

**Comment by rbaturov** [comment]
/cherry-pick release-4.18 release-4.17 release-4.16 release-4.15

File: test/e2e/performanceprofile/functests/1_performance/performance.go [hunk 3]
```diff
@@ -128,12 +128,16 @@ var _ = Describe("[rfe_id:27368][performance]", Ordered, func() {
...
-				isNodeBasedOnVM, err := infrastructure.IsVM(&node)
+				isNodeBasedOnVM, err := infrastructure.IsVM(context.TODO(), &node)
 				Expect(err).ToNot(HaveOccurred(), "Failed to detect if the node is based on VM")
-				if isNodeBasedOnVM {
-					testlog.Warning(fmt.Sprintf("Tuned profile is degraded. A warning raised as the node is based on a VM. Error message: %s", degradedCondition.Message))
-				} else {
-					Expect(degradedCondition.Status).To(Equal(corev1.ConditionFalse), "Tuned profile is degraded. Error message: %s", degradedCondition.Message)
+
+				if degradedCondition.Status == corev1.ConditionTrue {
+					message := fmt.Sprintf("Tuned profile is degraded. Error message: %s", degradedCondition.Message)
```
Comment (jmencak) [inline test/e2e/performanceprofile/functests/1_performance/performance.go:135]: I guess it might help which Tuned profile is degraded (better logging), but not a show stopper.

**Review by jmencak — APPROVED** [review]
/approve
/lgtm
to give a chance to others to review before merging, feel free to unhold
/hold
